website/blog/2020-06-16-alluxio-datacaching.md

```diff
@@ -0,0 +1,244 @@
+We also recorded **57%** improvement in IO for remote data source scans.
+
+
+
+# Presto Architecture
+
+Presto's architecture allows storage and computation to scale independently. However, scanning data from remote storage can be a costly operation, 
+and it makes achieving interactive query latency a challenge.
+
+Presto workers are responsible for executing query plan fragments on the data scanned from the independent and typically remote data sources. 
+Presto workers do not store any data for remote data sources which enables the computation to grow elastically.
+<br><br>
+The architecture diagram below highlights the data read paths from a remote HDFS source. 
+Each worker independently reads data from the remote data source. In this blog we will be only talking about optimizations done in the read 
+operations from the remote data source.
+<br><br>
+
+
+![](/img/blog/2020-06-16-alluxio-datacaching/presto_worker.jpg)
+
+<br>
+
+
+
+# Presto + Data Caching Architecture
+
+To solve sub-second latency use cases, we decided to implement various optimizations.  One important optimization was to implement a data cache. 
+Data caching has been a traditional optimization technique to bring the working dataset closer to the compute nodes and reduce trips to remote 
+storage to save latencies and IO.
+
+The challenge was to make the data caching effective when petabytes of data get scanned from the remote data sources with no fixed pattern. 
+Another requirement for data caching to be effective was to achieve data affinity in a distributed environment like Presto.
+<br><br>
+With the addition of data caching, the Presto architecture looks like the following:
+<br><br>
+
+![](/img/blog/2020-06-16-alluxio-datacaching/presto_worker_datacache.jpg)
+<br>
+
+More on this is covered in later sections.
+
+## Soft Affinity scheduling
+
+Presto’s current scheduler takes the worker load into account when distributing the splits, such scheduling strategy keeps the workload distribution uniform among workers. 
+But from the data locality perspective, it distributes splits randomly and not necessarily guarantees any affinity, which is required for any meaningful data caching effectiveness. 
+It is critical for the coordinator to leverage the same worker for a split which may contain the data for it in its cache.
+<br>
+
+![](/img/blog/2020-06-16-alluxio-datacaching/presto_affinity_scheduler.jpg)
+<br>
+
+The above diagram illustrates how affinity scheduling distributes various splits among the workers.
+<br><br>
+
+Soft affinity scheduling makes the best attempt to assign the same split to the same worker when doing the scheduling. The soft affinity scheduler uses the hash of a split to 
+choose a preferred worker for the split. Soft affinity scheduler:
+1. Computes a preferred worker for a spit. If the preferred worker has resources available then it is assigned the split.
```

Comment:
Accidentally wrote spit instead of split

Reply:
nice catch!